servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/RoundRobinToDefaultLBMigrationProvider.java

```diff
@@ -0,0 +1,154 @@
+import static io.servicetalk.utils.internal.NumberUtils.ensureNonNegative;
+
+public final class RoundRobinToDefaultLBMigrationProvider implements RoundRobinLoadBalancerBuilderProvider {
+
+    static final String PROPERTY_NAME = "io.servicetalk.loadbalancer.roundRobinUsesDefaultLoadBalancer";
+    private static final boolean DEFAULT_PROPERTY_VALUE = false;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RoundRobinToDefaultLBMigrationProvider.class);
+
+    private final Properties properties;
+
+    public RoundRobinToDefaultLBMigrationProvider() {
+        this(System.getProperties());
+    }
+
+    // Exposed for testing.
+    RoundRobinToDefaultLBMigrationProvider(final Properties properties) {
+        this.properties = properties;
+    }
+
+    @Override
+    public <ResolvedAddress, C extends LoadBalancedConnection>
+    RoundRobinLoadBalancerBuilder<ResolvedAddress, C> newBuilder(
+            String id, RoundRobinLoadBalancerBuilder<ResolvedAddress, C> builder) {
+        if (isEnabled()) {
+            LOGGER.info("Enabling DefaultLoadBalancer in place of RoundRobinLoadBalancer for client {}", id);
+            return new DefaultLoadBalancerRoundRobinBuilder<>(id);
+        } else {
+            LOGGER.debug("Not enabling DefaultLoadBalancer in place of RoundRobinLoadBalancer for client {}", id);
+            return builder;
+        }
+    }
+
+    private boolean isEnabled() {
+        String propertyValue = properties.getProperty(PROPERTY_NAME, Boolean.toString(DEFAULT_PROPERTY_VALUE));
+        return Boolean.TRUE.toString().equalsIgnoreCase(propertyValue);
+    }
+
+    private static final class DefaultLoadBalancerRoundRobinBuilder<ResolvedAddress, C extends LoadBalancedConnection>
+            implements RoundRobinLoadBalancerBuilder<ResolvedAddress, C> {
+
+        private final String id;
+        private int linearSearchSpace = 16;
+        @Nullable
+        private Executor backgroundExecutor;
+        private Duration healthCheckInterval = DEFAULT_HEALTH_CHECK_INTERVAL;
+        private Duration healthCheckJitter = DEFAULT_HEALTH_CHECK_JITTER;
+        private int healthCheckFailedConnectionsThreshold = DEFAULT_HEALTH_CHECK_FAILED_CONNECTIONS_THRESHOLD;
+        private Duration healthCheckResubscribeInterval = DEFAULT_HEALTH_CHECK_RESUBSCRIBE_INTERVAL;
+        private Duration healthCheckResubscribeJitter = DEFAULT_HEALTH_CHECK_JITTER;
+
+        DefaultLoadBalancerRoundRobinBuilder(final String id) {
+            this.id = id;
+        }
+
+        @Override
+        public RoundRobinLoadBalancerBuilder<ResolvedAddress, C> linearSearchSpace(int linearSearchSpace) {
+            this.linearSearchSpace = ensureNonNegative(linearSearchSpace, "linearSearchSpace");
+            return this;
+        }
+
+        @Override
+        public RoundRobinLoadBalancerBuilder<ResolvedAddress, C> backgroundExecutor(Executor backgroundExecutor) {
+            this.backgroundExecutor = backgroundExecutor;
```

Comment:
We have to wrap it with `NormalizedTimeSourceExecutor`

Reply:
That will be taken care of by the `DefaultLoadBalancerBuilder` called in the `build()` method.